subprojects/core-api/src/main/java/org/gradle/caching/package-info.java

```diff
@@ -19,7 +19,9 @@
  *
  * @since 3.3
  */
-@NonNullApi
-package org.gradle.caching;
 
-import org.gradle.api.NonNullApi;
+// We already have a similar package-info.java in :build-cache-spi
+// We must have a package-info.java for every package directory, but for each package
+// @NonNullApi must only be declared once, otherwise `:docs:javadocAll` fails.
+//@NonNullApi
```

Comment:
💅 Do we need commented out //@NonNullApi or we can remove this line?

Reply:
That's what the comment is about. Ideally, we would have the `@NonNullApi` annotation on this package, but we can't, because of the aforementioned reasons. I think having the "code" commented out makes sense. If you feel strongly, I'm happy to change it in the followup PR.